src/sdp.ts

```diff
@@ -15,6 +15,23 @@ const log = logger('libp2p:webrtc:sdp')
 // @ts-expect-error - Not easy to combine these types.
 export const mbdecoder: any = Object.values(bases).map(b => b.decoder).reduce((d, b) => d.or(b))
 
+export function getLocalFingerprint (pc: RTCPeerConnection): string | undefined {
+  const localDescription = pc.localDescription
+  if (localDescription == null) {
+    return undefined
+  }
+  return getFingerprintFromSdp(localDescription.sdp)
+}
+
+const fingerprintRegex = /^a=fingerprint:(?:\w+-[0-9]+)\s(?<fingerprint>(:?[0-9a-fA-F]{2})+)$/gm
+function getFingerprintFromSdp (sdp: string): string | undefined {
+  const searchResult = fingerprintRegex.exec(sdp)
+  if (searchResult == null) {
+    return ''
+  }
+
+  return searchResult.groups?.fingerprint
+}
```

Comment:
Prefer `regex.match`.

Using `exec` can cause unexpected issues with the stored `lastIndex` being updated and then causing match failures in later `.exec` calls. 

> JavaScript [RegExp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp) objects are stateful when they have the [global](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global) or [sticky](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/sticky) flags set (e.g. /foo/g or /foo/y). They store a [lastIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex) from the previous match. Using this internally, exec() can be used to iterate over multiple matches in a string of text (with capture groups), as opposed to getting just the matching strings with [String.prototype.match()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match).

see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#description

Also see https://stackoverflow.com/a/3811949/592760

Reply:
Thanks @SgtPooki . I was not aware of this. I've incorporated the change and added a simple test.